flink-autoscaler-plugin-jdbc/src/main/java/org/apache/flink/autoscaler/jdbc/event/JdbcAutoScalerEventHandler.java

```diff
@@ -104,6 +135,49 @@ public void handleScalingEvent(
         }
     }
 
+    @Override
+    public void close() {
+        if (Objects.nonNull(scheduledEventHandlerCleaner)
+                && !scheduledEventHandlerCleaner.isShutdown()) {
+            scheduledEventHandlerCleaner.shutdownNow();
+        }
+    }
+
+    @VisibleForTesting
+    void cleanExpiredEvents() {
+        var date =
+                Timestamp.from(
+                        jdbcEventInteractor
+                                .getCurrentInstant()
+                                .minusMillis(eventHandlerTtl.toMillis()));
+        try {
+            Long minId = jdbcEventInteractor.queryMinEventIdByCreateTime(date);
+            if (Objects.isNull(minId)) {
+                log.warn("No expired event handlers queried at {}", new Date());
+                return;
+            }
+            int deleted = 0, batch = 4098;
+            var count = 0L;
+            boolean cleanable = true;
+            for (long startId = minId; deleted == batch || cleanable; startId += batch) {
```

Comment:
1. IIUC, `deleted == batch` isn't needed. If they are not equal, you updated the cleanable, so we check cleanable is enough.
2. If 1 is acceptable, we don't need the `for loop`. `while(true) loop` is enough, and we can break the loop if `cleanable` is false.
3. If 1 is acceptable, `deleted` can be defined inside of `loop`.
4. If 2 is acceptable, we don't need `cleanable`. We can check `jdbcEventInteractor.queryMinEventIdByCreateTime(date) == null`, if it's true, break directly.

Reply:
Thank you very much for comments.

> IIUC, deleted == batch isn't needed. If they are not equal, you updated the cleanable, so we check cleanable is enough.

Yes, it does  overall. But here's a small difference in this condition.
The `deleted==batch` would reduce the sql query from the DB, 

`jdbcEventInteractor.queryMinEventIdByCreateTime(date)` This query function is more like searching for the starting point of an interval composed of consecutive IDs that meet the deletion criteria. the `deleted==batch` would drive the deletion action in the consecutive IDs.

Please correct me if I'm wrong.

Reply:
From the code, `jdbcEventInteractor.queryMinEventIdByCreateTime(date) != null;` is only called when `deleted!=batch`.

It means `jdbcEventInteractor.queryMinEventIdByCreateTime(date) != null;` won't be called when `deleted==batch`. And `for loop` will delete next round.

I don't understand why updating `deleted == batch || cleanable` to `cleanable` will query more times from Database.